### ISSUE TITLE
Use requestAnimationFrame

### ISSUE THREAD

**Issue by rivertam** [loktar00/react-lazy-load#54]
Putting functions as the scroll handler/resize handler blocks scrolling while they execute, so it's preferable to use `requestAnimationFrame` for the handlers.

**Comment by rivertam** [comment]
I'm planning on making a PR at some point if nobody else gets to it first. =)

**Comment by sergeylaptev** [comment]
Feel free to implement it, I'll check at the weekend

**Comment by sergeylaptev** [comment]
It would be great if you will split PR by issue. I guess that raf doesn't related with single scroll listener. Btw you can use https://www.npmjs.com/package/raf as polyfill

**Comment by rivertam** [comment]
Yeah, I just realized that I thought the part of the project that handled scroll handlers kind of belonged in a separate file, and I figured "if a separate file, why not a separate project"? Of course, by splitting it into a different project, I lost the ability to add them in piecewise. I think you'll find this approach actually produced more maintainable code; there are only 3 new lines to this repository aside from testing.

As for `raf`, I wanted to make the library I used have 0 dependencies. `raf` as a project spends more than half of its time concerned with cancelling requestAnimationFrame calls, which is not something this project or `manage-scroll-handlers` is concerned with. The only drawback to baking my own into the index file of `manage-scroll-handlers` is that it currently does not support vendor extensions. However, according to [caniuse.com](http://caniuse.com/#search=requestanimation), no browsers currently support  vendor extensions but *not* `requestAnimationFrame`, so almost all of `raf` is unnecessary for the project. I effectively reduced it to `requestAnimationFrame || (fn => setTimeout(fn, 30))`, which accomplishes the goal of making the callback non-blocking and in much fewer bytes. (one of my goals in the new project is to keep it under 2kb when minified)
